iGCS/Models/Craft/GCSCraftNotifications.m

```diff
@@ -24,4 +25,18 @@ + (void)didNavModeChangeFromLastHeartbeat:(mavlink_heartbeat_t) lastHeartbeat
     });
 }
 
++ (void)didMavModeChangeFromLastHeartbeat:(mavlink_heartbeat_t) lastHeartbeat
+                          andNewHeartbeat:(mavlink_heartbeat_t) newHeartbeat {
+
+    if (lastHeartbeat.base_mode != 0 &&
+        (lastHeartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED) == (newHeartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED)) {
```

Comment:
That's 3 places where this logic is repeated now.

Don't care if we use another utility method, or continue down the path of wrapping these into a heartbeat objects (which I would *happily* agree with :-), but need to deal with this. 

Also consider the lastHeartbeat.baseMode to be a little magical - how do we know that's not a valid value? Use another parameter to pass in the "first time" status, or (if going down the object route) encode this in the object.